tools/config-gen/params.yaml

```diff
@@ -15,49 +33,460 @@
+  deprecated: true
+  deprecation-warning: "This flag will be deleted soon."
 
 - flag-name: "log-severity"
   config-path: "logging.severity"
   type: "logSeverity"
-  default: "info"
   usage: "Specifies the logging severity expressed as one of [trace, debug, info, warning, error, off]"
+  default: "info"
 
-- flag-name: "client-protocol"
-  config-path: "gcs-connection.client-protocol"
-  type: "protocol"
-  default: "http1"
+- flag-name: "file-cache-max-size-mb"
+  config-path: "file-cache.max-size-mb"
+  type: "int"
+  usage: "Maximum size of the file-cache in MiBs"
+  default: "-1"
+
+- flag-name: "cache-file-for-range-read"
+  config-path: "file-cache.cache-file-for-range-read"
+  type: "bool"
+  usage: "Whether to cache file for range reads."
+  default: false
+
+- flag-name: "enable-crc-check"
+  config-path: "file-cache.enable-crc-check"
+  type: "bool"
+  usage: "Performs CRC check to ensure that file is correctly downloaded into cache."
+  default: true
+
+- flag-name: "enable-parallel-downloads"
+  config-path: "file-cache.enable-parallel-downloads"
+  type: "bool"
+  usage: "Enable parallel downloads."
+  default: false
+
+- flag-name: "download-parallelism-per-file"
+  config-path: "file-cache.download-parallelism-per-file"
+  type: "int"
+  usage: "Number of concurrent download requests per file."
+  default: "10"
+
+- flag-name: "max-download-parallelism"
+  config-path: "file-cache.max-download-parallelism"
+  type: "int"
+  usage: "Sets an uber limit of number of concurrent file download requests that are made across all files."
+  default: "-1"
+
+- flag-name: "read-request-size-mb"
+  config-path: "file-cache.read-request-size-mb"
+  type: "int"
+  usage: "Size of chunks in MiB that each concurrent request downloads."
+  default: "0"
+
+- flag-name: "cache-dir"
+  config-path: "cache-dir"
+  type: "resolvedPath"
+  usage: "Enables file-caching. Specifies the directory to use for file-cache."
+
+- flag-name: "enable-empty-managed-folders"
+  config-path: "list.enable-empty-managed-folders"
+  type: "bool"
   usage: >-
-    The protocol used for communicating with the GCS backend.
-    Value can be 'http1' (HTTP/1.1), 'http2' (HTTP/2) or 'grpc'.
+    This handles the corner case in listing managed folders.
+    There are two corner cases (a) empty managed folder (b) nested managed folder which doesn't contain any descendent as object.
+    This flag always works in conjunction with --implicit-dirs flag.
+    (a) If only ImplicitDirectories is true, all managed folders are listed other than above two mentioned cases.
+    (b) If both ImplicitDirectories and EnableEmptyManagedFolders are true, then all the managed folders are listed including the above-mentioned corner case.
+    (c) If ImplicitDirectories is false then no managed folders are listed irrespective of enable-empty-managed-folders flag.
+  default: false
+
+- flag-name: "experimental-grpc-conn-pool-size"
+  config-path: "gcs-connection.grpc-conn-pool-size"
```

Comment:
should this also be experimental

Reply:
I don't think so. This config currently exists (and is not a new config). So, renaming it is out of scope for this PR.